### PR TITLE
feat(authenticator, m3): update authenticator for material 3

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:*

*Description of changes:*
- use `FilledButton` for primary buttons when material 3 is enabled
- use `SearchAnchor`for country code selector when material 3 is enabled
- correctly alphabetize countries (not material 3 specific)
- add material 3 to UI tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/authenticator/amplify_authenticator/test/ui/main_test.dart [hunk 1]
```diff
@@ -49,22 +49,28 @@ enum ScreenGeometry {
   final double pixelRatio;
 }
 
+// TODO(Jordan-Nelson): Update all themes to use material3
```
Comment (Equartey) [inline packages/authenticator/amplify_authenticator/test/ui/main_test.dart:52]: Is this still needed?

**Review by Equartey — APPROVED** [review]

**Review by NikaHsn — APPROVED** [review]

File: packages/authenticator/amplify_authenticator/example/lib/customization/authenticator_with_custom_auth_flow.dart [hunk 2]
```diff
@@ -52,20 +52,8 @@ class AuthenticatorWithCustomAuthFlow extends StatelessWidget {
         }
       },
       child: MaterialApp(
-        theme: ThemeData.from(
```
Comment (dnys1) [inline packages/authenticator/amplify_authenticator/example/lib/customization/authenticator_with_custom_auth_flow.dart:55]: Should we have an example which showcases a custom Mat3 theme?

File: packages/authenticator/amplify_authenticator/test/ui/goldens/theme_socialProviderConfig_signInStep_material3Theme_lightMode_laptopGeometry.png
Comment (dnys1) [inline packages/authenticator/amplify_authenticator/test/ui/goldens/theme_socialProviderConfig_signInStep_material3Theme_lightMode_laptopGeometry.png:1]: Is the outline correct?

Comment (Jordan-Nelson) [reply packages/authenticator/amplify_authenticator/test/ui/goldens/theme_socialProviderConfig_signInStep_material3Theme_lightMode_laptopGeometry.png:1]: I'm not sure why it shows such a dark outline here. The tests might not be accurately displaying the shadow from the Material Card. I think it is fine that the tests are not 100% pixel perfect as long as they are consistent.

This is what is actually looks like.

#### MacOS

Screenshot 2023-05-19 at 3 14 14 PM

#### Web

Screenshot 2023-05-19 at 3 15 25 PM

**Pull request merged**